cpp/src/arrow/dataset/partition_test.cc

```diff
@@ -426,6 +456,41 @@ TEST_F(TestPartitioning, HivePartitioning) {
   AssertParseError("/alpha=0.0/beta=3.25/");  // conversion of "0.0" to int32 fails
 }
 
+TEST_F(TestPartitioning, HivePartitioningEquals) {
+  const auto& array_vector = ArrayVector();
+  std::vector<std::shared_ptr<arrow::Array>> other_vector(2);
+  other_vector.push_back(ArrayFromJSON(utf8(), R"(["foo", "bar", "baz"])"));
+  other_vector.push_back(ArrayFromJSON(utf8(), R"(["bar", "foo", "baz"])"));
```

Comment:
Er, this is creating a vector of size 4 with the first two values zero-initialized, AFAIU.
```suggestion
  ArrayVector other_vector(2);
  other_vector[0] = ArrayFromJSON(utf8(), R"(["foo", "bar", "baz"])");
  other_vector[1] = ArrayFromJSON(utf8(), R"(["bar", "foo", "baz"])");
```